File: backend/modules/sabor-familia/src/main/java/br/com/puc/saborfamilia/service/personalizacao/PersonalizacaoService.java
package br.com.puc.saborfamilia.service.personalizacao;

import br.com.puc.saborfamilia.database.entity.PersonalizacaoEntity;
import br.com.puc.saborfamilia.service.personalizacao.dto.request.EditarPersonalizacaoRequest;
import br.com.puc.saborfamilia.service.personalizacao.dto.response.CatalogoPersonalizacaoResponse;
import br.com.puc.saborfamilia.service.personalizacao.dto.response.PersonalizacaoResponse;
import br.com.puc.saborfamilia.service.personalizacao.dto.response.PersonalizacaoResumoResponse;
import java.util.List;

public interface PersonalizacaoService {

  List<CatalogoPersonalizacaoResponse> listarCatalogo();

  List<PersonalizacaoResumoResponse> listarCatalogoContextoPerfil();

  List<PersonalizacaoResumoResponse> listarCatalogoContextoReceita();

  PersonalizacaoResponse editarPersonalizacao(String codigo, EditarPersonalizacaoRequest request);

  PersonalizacaoResponse ativarPersonalizacao(String codigo);

  PersonalizacaoResponse inativarPersonalizacao(String codigo);

  List<PersonalizacaoEntity> validarPersonalizacoes(List<String> codigosPersonalizacao);

}

File: frontend/src/features/sabor_familia/service/PersonalizacaoService.ts
import api from "../lib/Api";
import type { CatalogoPersonalizacaoResponse } from "../dto/personalizacao/response/CatalogoPersonalizacaoResponse";
import type { PersonalizacaoResumoResponse } from "../dto/personalizacao/response/PersonalizacaoResumoResponse";

export const personalizacaoService = {
  listarCatalogo: async (): Promise<CatalogoPersonalizacaoResponse[]> => {
    const { data } = await api.get<CatalogoPersonalizacaoResponse[]>("/personalizacao");
    return data;
  },

  listarCatalogoContextoReceita: async (): Promise<PersonalizacaoResumoResponse[]> => {
    const { data } = await api.get<PersonalizacaoResumoResponse[]>("/personalizacao/receita");
    return data;
  },
};
